lib/galaxy/tools/__init__.py

```diff
@@ -2541,7 +2541,7 @@ def iterate(group_inputs, inputs, state_inputs, other_values=None):
             other_values = ExpressionContext( state_inputs, other_values )
             for input_index, input in enumerate( inputs.itervalues() ):
                 # create model dictionary
-                tool_dict = input.to_dict(trans)
+                tool_dict = input.to_dict(trans, other_values=other_values)
```

Comment:
This breaks grouping elements i.e. repeats, conditionals. The other changes should have no effect.